verde/coordinates.py

```diff
@@ -1005,6 +1003,125 @@ def _check_rolling_window_overlap(region, size, shape, spacing):
         )
 
 
+def expanding_window(coordinates, center, sizes):
+    """
+    Select points on windows of changing size around a center point.
+
+    Returns the indices of points falling inside each window.
+
+    Parameters
+    ----------
+    coordinates : tuple of arrays
+        Arrays with the coordinates of each data point. Should be in the
+        following order: (easting, northing, vertical, ...).
+    center : tuple
+        The coordinates of the center of the window. Should be in the
+        following order: (easting, northing, vertical, ...).
+    sizes : array
+        The sizes of the windows. Does not have to be in any particular order.
+        The order of indices returned will match the order of window sizes
+        given. Units should match the units of *coordinates* and *center*.
+
+    Returns
+    -------
+    indices : list
+        Each element of the list corresponds to  the indices of points falling
+        inside a window. Use them to index the coordinates for each window. The
+        indices will depend on the number of dimensions in the input
+        coordinates. For example, if the coordinates are 2D arrays, each window
+        will contain indices for 2 dimensions (row, column).
+
+    See also
+    --------
+    block_split : Split a region into blocks and label points accordingly.
+    rolling_window : Select points on a rolling (moving) window.
+
+    Examples
```

Comment:
The example looks great. Easy to follow and understand how the expanding window works on 1D and 2D data